client/my-sites/plugins/jetpack-plugins-setup/index.jsx

```diff
@@ -263,73 +267,94 @@ const PlansSetup = React.createClass( {
+		}
+	},
+
+	renderStatusError( plugin ) {
+		const { translate } = this.props;
+
+		// This state isn't quite an error
+		if ( plugin.error.code === 'already_registered' ) {
+			return (
+				<Notice
+					showDismiss={ false }
+					isCompact={ true }
+					status="is-info"
+					text={ translate( 'This plugin is already registered with another plan.' ) }
+				>
+					<NoticeAction key="notice_action" href="/me/purchases" onClick={ this.trackManagePlans }>
+						{ translate( 'Manage Plans' ) }
+					</NoticeAction>
+				</Notice>
+			);
+		}
+
+		const statusProps = {
+			isCompact: true,
+			status: 'is-error',
+			showDismiss: false,
+		};
+		statusProps.children = (
+			<NoticeAction key="notice_action" href={ helpLinks[ plugin.slug ] } onClick={ this.trackManualInstall }>
+				{ translate( 'Manual Installation' ) }
+			</NoticeAction>
+		);
+
+		switch ( plugin.status ) {
+			case 'install':
+				return (
+					<Notice { ...statusProps } text={ translate(
 						'An error occurred when installing %(plugin)s.',
 						{ args: { plugin: plugin.name } }
-					);
-					break;
-				case 'activate':
-					statusProps.text = translate(
+					) } />
+				);
+			case 'activate':
+				return (
+					<Notice { ...statusProps } text={ translate(
 						'An error occurred when activating %(plugin)s.',
 						{ args: { plugin: plugin.name } }
-					);
-					break;
-				case 'configure':
-					statusProps.text = translate(
+					) } />
+				);
+			case 'configure':
+				return (
+					<Notice { ...statusProps } text={ translate(
 						'An error occurred when configuring %(plugin)s.',
 						{ args: { plugin: plugin.name } }
-					);
-					break;
-				default:
-					statusProps.text = plugin.error.message || translate( 'An error occured.' );
-					break;
-			}
-			statusProps.children = (
-				<NoticeAction key="notice_action" href={ helpLinks[ plugin.slug ] } onClick={ this.trackManualInstall }>
-					{ translate( 'Manual Installation' ) }
-				</NoticeAction>
-			);
-			if ( plugin.error.code === 'already_registered' ) {
-				statusProps.status = 'is-info';
-				statusProps.text = translate( 'This plugin is already registered with another plan.' );
-				delete statusProps.children;
-			}
-		} else {
-			statusProps.icon = 'plugins';
-			statusProps.status = 'is-info';
-			switch ( plugin.status ) {
-				case 'done':
-					// Done doesn't use a notice
-					return (
-						<div className="plugin-item__finished">
-							{ translate( 'Successfully installed & configured.' ) }
-						</div>
-					);
-				case 'activate':
-				case 'configure':
-					statusProps.text = translate( 'Almost done' );
-					break;
-				case 'install':
-					statusProps.text = translate( 'Working…' );
-					break;
-				case 'wait':
-				default:
-					statusProps.text = translate( 'Waiting to install' );
-			}
+					) } />
+				);
+			default:
+				return (
+					<Notice { ...statusProps } text={ plugin.error.message || translate( 'An error occured.' ) } />
```

Comment:
I'm not familiar with the endpoint, but unless the error messages are intended to be seen by a user, these can end up being rather cryptic for folks. It also avoids very funny cases where we might have a huge trace in the message. Anyway pretty minor, but I try to avoid these.